src/smefit/chi2.py

```diff
@@ -217,11 +301,22 @@ def compute_scan(self):
             for coeff in self.coefficients:
                 if coeff.name not in self.chi2_dict:
                     continue
-                self.chi2_dict[coeff.name][rep] = self.regularized_chi2_func(
-                    coeff, self.chi2_dict[coeff.name]["x"], use_replica
-                )
+                if self.mass_scan:
+                    self.chi2_dict[coeff.name][rep] = self.chi2_mass_scan(
+                        coeff, self.chi2_dict[coeff.name]["x"]
+                    )
+                else:
+                    self.chi2_dict[coeff.name][rep] = self.regularized_chi2_func(
+                        coeff, self.chi2_dict[coeff.name]["x"], use_replica
+                    )
+                # Parameter is scanned,
+                # reset to 0 to not affect the next scan
                 coeff.value = 0.0
-                coeff.is_free = False
```

Comment:
@jacoterh @giacomomagni do you know why this was here? I didn't understand it and it seems to be working without it? Or maybe i missed something

Reply:
Maybe it is needed for the next iterations, like the comment says?

Reply:
I don't see why...